### PR TITLE
Reports overview

### PR DESCRIPTION
**Pull request by kaapstorm**
First draft at a doc.

### CODE REVIEW

File: corehq/apps/userreports/reports/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+There are two main components involved in producing a report:
+
+* the data source configuration
+* the report configuration
+
+
+The data source
+---------------
+
+The data source is usually a table in PostgreSQL, but data sources can also be ElasticSearch indexes.
```
Comment (emord) [inline corehq/apps/userreports/reports/README.md]: ES is only used by 1 data source in enikshay. I'd probably leave it out of these docs as I don't think it's every going to be used (and if it is that person should know what they're doing)

File: corehq/apps/userreports/reports/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+A data source is defined in a DataSourceConfiguration instance. This determines where its data comes from (cases or forms), and how the data is filtered. The data source is populated by a background task. Data sources are not aggregated.
```
Comment (emord) [inline corehq/apps/userreports/reports/README.md]: > Data sources are not aggregated.

Perhaps it should say that there are generally 1 (sometimes, but rarely more) row per case/form?

**Review by emord — COMMENTED** [review]
Not sure what who the audience is for this (guessing devs / @gbova ? )

Looked like there is some overlap with the main UCR docs in the folder above (which is a huge doc that I want to split up at some point). Is this just supposed to be a quick over view of report builder?

**Review by gbova — APPROVED** [review]
Makes sense to me! And @emord I think you're right, devs will be the only ones looking at this.

File: corehq/apps/userreports/reports/README.md (hunk 1 above)
Comment (kaapstorm) [reply corehq/apps/userreports/reports/README.md]: Good to know. I had only come across Postgres data sources, but I saw the code supported ES.

**Comment by kaapstorm** [comment]
Yes, @emord I was looking for something like a primer for how reports work, and I didn't find anything that didn't assume an introductory level of knowledge already. I've used the UCR docs a lot, and I found the Report Builder readme, but they didn't quite hit the spot. 

This was pretty hastily thrown together. I'm sure these docs can be structured better, to work together. I imagine if we treat documentation a bit more like coding, I could do a bit of refactoring before I add this. I'm open to suggestions. 

And if you've got some ideas about what to do with the UCR docs, let me know if there is a way to edit / chop up / merge this to add to them. Otherwise I can take a shot at it, and ping you when I'm done.

**Comment by emord** [comment]
@kaapstorm I don't have a system that I want to use for the UCR docs just noting that these docs have been a pain point for me in the past. It does seem useful to have a general report builder readme. 

This looks good from my end. I think you can feel free to self merge when you're happy with it since it's only docs

**Pull request merged**